File: 3-two_sum/two_sum.py
# By using nested loop and By Brute Force approach 
# def two_sum(self, arr, target):
#     for i in range(len(arr)):
#         for j in range(i+1 , len(arr)):
#             if arr[i] + arr[j] == target:
#                 return [i,j]
            

# By using hashmap and checking values like  target - arr[i] and check that answer from the hashmap

def two_sum(self, arr, target):
    seen = {}
    for i,num in enumerate(arr):
        comp = target - num
        if comp in seen:
            return [seen[comp],i]
        seen[num] = i

File: 3-two_sum/two_sum.java
public class two_sum{
    public class static int[] twoSum(int[] nums, target){

        // with the help of nested loop and by brute force approach

        // for(int i = 0; i<nums.length; i++){
        //     for(int j=i+1; j<nums.length; j++){
        //         if(nums[i] + nums[j] == target){
        //             return new int[]{i,j}
        //         }
        //     }
        // }
        // return null

        HashMap<Integer> seen = HashMap<>();
        for(int i = 0; i<nums.length;i++){
            int comp = target - nums[i];
            if(seen.containsKay(com)){
                return new int[]{map.get(comp), i};
            }
            seen.put(nums[i],i)
        }
    }
}